shell/src/test/scala/ammonite/shell/SessionTests.scala

```diff
@@ -81,12 +81,55 @@ object SessionTests extends TestSuite{
 
         @ cd! 'destSymLink
 
-        @ assert("srcDir0" == wd.name)
+        @ assert("srcDir0" == wd.tryFollowLinks.get.name)
+
+        @ assert("destSymLink" == wd.name)
 
         @ cd! originalWd
 
         @ rm! tmpdir
       """)
     }
+
+    'nestedSymlinks {
+      check.session(
+        s"""
+        @ import ammonite.ops._
+
+        @ interp.load.module($bareSrc)
+
+        @ val originalWd = wd
+
+        @ val names = Seq('test123, 'test124, 'test125, 'test126)
+
+        @ names.foreach(p => rm! wd/p)
+
+        @ mkdir! wd/'test123
+
+        @ ln.s(wd/'test123, wd/'test124)
+
+        @ ln.s(wd/'test124, wd/'test125)
+
+        @ ln.s(wd/'test125, wd/'test126)
+
+        @ cd! 'test126
+
+        @ assert(wd.tryFollowLinks.get == originalWd/'test123)
+
+        @ assert(wd == originalWd/'test126)
+
+        @ cd! originalWd
+
+        @ assert(wd == originalWd)
+
+        @ rm! 'test123
+
+        @ assert( (wd / 'test126).tryFollowLinks == None )
```

Comment:
Ideally, here I want to try `cd! wd/'test126` and catch an exception. But I don't know how to specify catching an exception in this test format.